src/vm/wks/CMakeLists.txt

```diff
@@ -105,11 +105,15 @@ if (WIN32)
     endif()
 
     add_custom_command(
-        # The AsmConstants.inc will be built in the pre-build phase of the cee_wks build
-        TARGET cee_wks PRE_BUILD 
+        OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/AsmConstants.inc"
+        DEPENDS ${VM_DIR}/${ARCH_SOURCES_DIR}/asmconstants.h
         COMMAND ${POWERSHELL} -NoProfile -ExecutionPolicy Bypass -NonInteractive \"& \"\"${VM_DIR}/h2inc.ps1\"\"\" \"\"\"${VM_DIR}/${ARCH_SOURCES_DIR}/asmconstants.h\"\"\" >"${CMAKE_CURRENT_BINARY_DIR}/AsmConstants.tmp"
         COMMAND ${CMAKE_CXX_COMPILER} ${DEFINITIONS} /EP "${CMAKE_CURRENT_BINARY_DIR}/AsmConstants.tmp" >"${CMAKE_CURRENT_BINARY_DIR}/AsmConstants.inc"
     )
+
+    set_source_files_properties(${CMAKE_CURRENT_BINARY_DIR}/AsmConstants.inc PROPERTIES GENERATED TRUE)
+    set_property(SOURCE ${VM_SOURCES_WKS_ARCH_ASM} PROPERTY OBJECT_DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/AsmConstants.inc)
```

Comment:
This is not enough to create a proper dependency, unless the object_depends set for all the sources is sufficient, which I am slightly not sure about - the cmake doc says it is a deprecated way. The way we use everywhere else is to add custom target, make it depend on the asmconstants.inc and make the targets that are built out of the asm files that include the asmconstants.inc  depend on that custom target. Unless cmake has problems detecting the dependency of the asm files on the inc file, that should be sufficient.
Like:
```cmake
 add_custom_target(
    asmconstants_inc
    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/AsmConstants.inc
  )
```